slasher/detection/filter/filter_test.go

```diff
@@ -0,0 +1,208 @@
+package filter
+
+import (
+	"testing"
+
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/go-ssz"
+)
+
+func (f Filter) String() string {
+	s := make([]byte, 8*len(f))
+	for i, x := range f {
+		for j := 0; j < 8; j++ {
+			if x&(1<<uint(j)) != 0 {
+				s[8*i+j] = '1'
+			} else {
+				s[8*i+j] = '.'
+			}
+		}
+	}
+	return string(s)
+}
+
+func TestFilter_OK(t *testing.T) {
+	attData := &ethpb.AttestationData{
+		Slot:           4,
+		CommitteeIndex: 2,
+		Target: &ethpb.Checkpoint{
+			Epoch: 4,
+			Root:  []byte("wow"),
+		},
+		Source: &ethpb.Checkpoint{
+			Epoch: 3,
+			Root:  []byte("eth2"),
+		},
+		BeaconBlockRoot: []byte("is great"),
+	}
+	dataRoot, err := ssz.HashTreeRoot(attData)
+	if err != nil {
+		t.Fatal(err)
+	}
+	f, err := NewFilter(dataRoot[:])
+	if err != nil {
+		t.Fatal(err)
+	}
+	got := f.String()
+	want := "...1.....1.1.11."
+	if got != want {
+		t.Fatalf("got %s, want %s", got, want)
+	}
+
+	found, err := f.Contains(dataRoot[:])
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !found {
+		t.Error("Did not expect filter to not contain entered data")
+	}
+
+	diffAttData := &ethpb.AttestationData{
+		Slot:           4,
+		CommitteeIndex: 2,
+		Target: &ethpb.Checkpoint{
+			Epoch: 4,
+			Root:  []byte("wowzers"),
+		},
+		Source: &ethpb.Checkpoint{
+			Epoch: 3,
+			Root:  []byte("eth2.0"),
+		},
+		BeaconBlockRoot: []byte("is really great"),
+	}
+	diffDataRoot, err := ssz.HashTreeRoot(diffAttData)
+	if err != nil {
+		t.Fatal(err)
+	}
+	found, err = f.Contains(diffDataRoot[:])
+	if err != nil {
+		t.Fatal(err)
+	}
+	if found {
+		t.Error("Did not expect filter to contain non-existent root")
+	}
+
+	diffAttData = &ethpb.AttestationData{
+		Slot:           4,
+		CommitteeIndex: 2,
+		Target: &ethpb.Checkpoint{
+			Epoch: 4,
+			Root:  []byte("hahaha"),
+		},
+		Source: &ethpb.Checkpoint{
+			Epoch: 3,
+			Root:  []byte("i love"),
+		},
+		BeaconBlockRoot: []byte("eth2"),
+	}
+	diffDataRoot, err = ssz.HashTreeRoot(diffAttData)
+	if err != nil {
+		t.Fatal(err)
+	}
+	found, err = f.Contains(diffDataRoot[:])
+	if err != nil {
+		t.Fatal(err)
+	}
+	if found {
+		t.Error("Did not expect filter to contain non-existent root")
+	}
+}
+
+func TestFilter_NoCollisions(t *testing.T) {
```

Comment:
Can we instead have table driven tests that can easily test a bunch of different cases (single element, various elements, collisions, no collisions, etc.)